src/utils/debounce.ts

```diff
@@ -0,0 +1,26 @@
+
+export function debounce<T extends (...args: any[]) => void>(fn: T, wait?: number, leading?: boolean) {
+  let timer: number | NodeJS.Timeout;
+  let lastCalledAt = -Infinity;
+
+
+  return function debouncedFn(this: unknown, ...args: Parameters<T>) {
+    if (leading) {
+      const now = Date.now();
+      const elapsed = now - lastCalledAt;
+
+      lastCalledAt = now;
+
+      if (elapsed >= wait!) {
+        fn.apply(this, args);
+      }
+      return;
```

Comment:
Maybe it's better to use an `else` block here because this `if` block is quite long.